batchflow/notifier.py

```diff
@@ -430,7 +431,8 @@ def update_plot(self, index=0, add_suptitle=False, savepath=None, clear_display=
             self.plotter.config['suptitle'] = self.bar.format_meter(**fmt)
             self.plotter.annotate()
 
-        for i, container in enumerate(self.data_containers):
+        data_containers = [container for container in self.data_containers if container['data'] is not None]
+        for i, container in enumerate(data_containers):
             if i >= index:
                 subplot_index = i - index
                 subplot_config = plot_config.maybe_index(subplot_index)
```

Comment:
Why you use comprehension and cycle instead of skipping wrong cntainer in cycle? Looks like double itration.
Proposal:
```Python
for i, container in enumerate(self.data_containers):
    if container['data'] is None:
        continue
```

Reply:
Skipping the items in the same loop would make it so that `i` variable is also skipped (and, therefore, incorrect)